### PR TITLE
[receiver/haproxyreceiver] add more metrics

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Add a lot more metrics to the haproxyreceiver. Use attributes as resource attributes if possible.

**Link to tracking Issue:**
#16829

**Testing:**
Unit tests.

**Documentation:**
Generated from yaml.

### CODE REVIEW

File: receiver/haproxyreceiver/metadata.yaml [hunk 1]
```diff
@@ -22,49 +22,241 @@ resource_attributes:
...
   proxy_name:
     description: Proxy name
     type: string
   service_name:
     description: Service name (FRONTEND for frontend, BACKEND for backend, any name for server/listener)
     type: string
 
+attributes:
+  status_code:
+    description: Status code category, 1xx, 2xx, 3xx, 4xx, 5xx or other
```
Comment (MovieStoreGuy) [inline receiver/haproxyreceiver/metadata.yaml:34]: You can set this to be an enum value from memory. 

I think the structure is:

```yaml
attributes:
   status_code:
      description: Status code group
      type: string
      enum:
      - "1xx"
      - "2xx"
      - "3xx"
      - "4xx"
      - "5xx"
```

File: receiver/haproxyreceiver/metadata.yaml [hunk 2]
```diff
@@ -22,49 +22,241 @@ resource_attributes:
...
+    type: string
+
 metrics:
-  haproxy.connection_rate:
+  haproxy.connections.rate:
```
Comment (MovieStoreGuy) [inline receiver/haproxyreceiver/metadata.yaml:38]: I appreciate the move to `.` spacing, thank you.

File: receiver/haproxyreceiver/metadata.yaml [hunk 3]
```diff
@@ -22,49 +22,241 @@ resource_attributes:
...
-    description: Total number of requests on this worker process since started. Corresponds to HAProxy's `CumReq` metric.
+  haproxy.sessions.count:
+    description: Current sessions. Corresponds to HAProxy's `scur` metric.
+    enabled: true
+    gauge:
+      value_type: int
+      input_type: string
+    unit: "{sessions}"
+  haproxy.connections.total:
+    description: Cumulative number of connections (frontend). Corresponds to HAProxy's `conn_tot` metric.
```
Comment (MovieStoreGuy) [inline receiver/haproxyreceiver/metadata.yaml:53]: Is it worth referencing the HA Proxy's documentation for those who want to read further about the original / underlying metric using the `extended_documentation` field?

**Review by MovieStoreGuy — COMMENTED** [review]
Loving the increased addition of metrics, however, instead of using the description to highlight the relation to the actual HAProxy metric, lets use the `extended_documentation` field to link back to the HA Proxy docs so that any and all possible questions could be explained about what the original value means / comes from.

**Comment by atoulme** [comment]
Thanks for the early review! Much appreciated. I was pushing last night and updated dependencies when I saw there was a conflict - it looks like the new dependencies introduced this new explicit error for empty value. I have fixed those now. 

Agree on linking to haproxy, but not sure on a canonical place. I will have to look.

File: receiver/haproxyreceiver/metadata.yaml (hunk 3 above)
Comment (atoulme) [reply receiver/haproxyreceiver/metadata.yaml:61]: I can't find an authoritative documentation for haproxy metrics :(

**Comment by atoulme** [comment]
I have not found a source of truth for haproxy metrics. Can we continue the review and revisit later?

**Review by MovieStoreGuy — APPROVED** [review]

**Comment by bogdandrutu** [comment]
Needs rebase

**Pull request merged**